File: Strata/trunk/java/tests/StrataCommonTest/src/strata1/common/task/TestTaskSelector.java
// ##########################################################################
// # File Name:	TestTaskSelector.java
// #
// # Copyright:	2014, Sapientia Systems, LLC. All Rights Reserved.
// #
// # License:	This file is part of the StrataCommonTest Framework.
// #
// #   			The StrataCommonTest Framework is free software: you 
// #			can redistribute it and/or modify it under the terms of 
// #			the GNU Lesser General Public License as published by
// #    		the Free Software Foundation, either version 3 of the 
// #			License, or (at your option) any later version.
// #
// #    		The StrataCommonTest Framework is distributed in the 
// #			hope that it will be useful, but WITHOUT ANY WARRANTY; 
// #			without even the implied warranty of MERCHANTABILITY or 
// #			FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser 
// #			General Public License for more details.
// #
// #    		You should have received a copy of the GNU Lesser 
// #			General Public License along with the StrataCommonTest
// #			Framework. If not, see http://www.gnu.org/licenses/.
// ##########################################################################

package strata1.common.task;

/****************************************************************************
 * 
 * @author 		
 *     Sapientia Systems
 * @conventions	
 *     <a href="{@docRoot}/NamingConventions.html">Naming Conventions</a>
 */
public 
class TestTaskSelector
    implements ITaskSelector
{
    private int itsTaskId;
    
    /************************************************************************
     * Creates a new {@code TestTaskSelector}. 
     *
     */
    public 
    TestTaskSelector(int taskId)
    {
        itsTaskId = taskId;
    }

    /************************************************************************
     * {@inheritDoc} 
     */
    @Override
    public boolean 
    match(ITask task)
    {
        if ( task.hasProperty( Integer.class,"taskId" ) )
            return 
                task
                    .getProperty( Integer.class,"taskId" )
                    .equals(itsTaskId);
            
        return false;
    }

}

// ##########################################################################

File: csharp/tests/StrataCommonTest/Task/TestTaskSelector.cs
//  ##########################################################################
//  # File Name: TestTaskSelector.cs
//  # Copyright: 2011-2018, Sapientia Systems, LLC.
//  ##########################################################################
 
using System;
using System.Collections.Generic;

namespace Strata.Common.Task
{
    //////////////////////////////////////////////////////////////////////////
    /// <summary>
    /// $comments$
    /// </summary>
    ///  
    public class TestTaskSelector:
        ITaskSelector,
        IEquatable<ITaskSelector>
    {
        private int taskId;

        public
        TestTaskSelector(int tid)
        {
            taskId = tid;
        }

        public bool 
        Match(ITask task)
        {
            if ( task.HasProperty<int>( "TaskId" ) )
	            return task.GetProperty<int>( "TaskId" ) == taskId;
	            
	        return false;
        }

        public bool 
        Equals(ITaskSelector other)
        {
            if ( other is TestTaskSelector )
                return taskId == ((TestTaskSelector)other).taskId;

            return false;
        }

        public override int
        GetHashCode()
        {
            return 37 * taskId.GetHashCode(); 
        }
        
    }
}

//  ##########################################################################
